cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +70,37 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def find_reachable_deps(yarn_lock, dep):
+    """
+    Get a set of all dependencies reachable by a top-level dependency using BFS.
+
+    :param yarn_lock: yarn.lock file as a dictionary
+    :param dep: top-level dependency in package.json
+    :return: set of all dependencies reachable by dep
+    :rtype: set
+    """
+    reachable_deps = set()
+    bfs_queue = deque()
+    reachable_deps.add(dep)
+    bfs_queue.append(dep)
+    while bfs_queue:
+        current_dep = bfs_queue.popleft()
+        for dep_identifier, dep_data in yarn_lock.items():
+            package = pyarn.lockfile.Package.from_dict(dep_identifier, dep_data)
+            if current_dep in dep_identifier:
```

Comment:
Can this do false positive matching?

Reply:
Not as far as I can see. This may require more research. @chmeliik , what do you think?

Reply:
Proper handling would be much nicer, you can split the multi-keys using something like this: https://github.com/release-engineering/cachito/blob/8091e35befb9026e43069faa7c11c01e20279398/cachito/workers/pkg_managers/yarn.py#L287

Reply:
Done!